release_notes/markdown.py

```diff
@@ -0,0 +1,152 @@
+import dataclasses
+import logging
+import typing
+from collections import defaultdict
+
+import release_notes.model as rnm
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Title:
+    display: str
+    identifiers: list[str]
+
+    def __hash__(self):
+        return hash(self.display)
+
+    def __eq__(self, other):
+        return (
+            other is not None
+            and isinstance(other, Title)
+            and (other == self or hash(other) == hash(self))
+        )
+
+
+def _simple_title(display: str) -> Title:
+    return Title(display, [display.lower()])
+
+
+categories: list[Title] = [
+    Title('⚠️ Breaking Changes', ['action', 'breaking']),
+    Title('📰 Noteworthy', ['noteworthy']),
+    Title('🏃 Others', ['improvement', 'other']),
+    Title('✨ New Features', ['feature']),
+    Title('🐛 Bug Fixes', ['bugfix', 'fix']),
+    Title('📖 Documentation', ['doc']),
+]
+
+target_groups: list[Title] = [
+    _simple_title('USER'),
+    _simple_title('OPERATOR'),
+    _simple_title('DEVELOPER'),
+    _simple_title('DEPENDENCY'),
+]
+
+# identifier -> Title
```

Comment:
codestyle: how about using a type-alias for str, like so:
```
identifier = str
```

and then using `dict[identifier, Title]` as typehint.